core/transactions/transactions_cleanup.cxx

```diff
@@ -526,7 +517,7 @@ transactions_cleanup::attempts_loop()
 void
 transactions_cleanup::add_attempt(attempt_context& ctx)
 {
-    auto& ctx_impl = static_cast<attempt_context_impl&>(ctx);
+    auto& ctx_impl = dynamic_cast<attempt_context_impl&>(ctx);
```

Comment:
What is the reason if this change? I think it is exactly one kind of derived class here, right? I wonder if we really need this dynamic_cast, which can do extra checks in runtime and might return `nullptr`.

https://en.cppreference.com/w/cpp/language/dynamic_cast
> If the cast is successful, dynamic_cast returns a value of type new-type. If the cast fails and new-type is a pointer type, it returns a null pointer of that type. If the cast fails and new-type is a reference type, it throws an exception that matches a handler of type [std::bad_cast](https://en.cppreference.com/w/cpp/types/bad_cast).

Reply:
Oh -- IIRC CLion had suggested it.  I never check the result anyways, as this has to succeed given the class hierarchy.   Ya back to static_cast.